DOCS-7946: Add instructions for custom frameworks (#23673)
DataDog/documentation: content/en/security/cloud_security_management/misconfigurations/frameworks_and_benchmarks/custom_frameworks.md

  text: "Search and explore misconfigurations"
---

With custom frameworks, you can define and measure compliance against your own cloud security baseline. Custom frameworks are listed on the Cloud Security Management (CSM) [Compliance][6] page, have their own real-time report and [security posture score][7], and are queryable within explorers and dashboards.

1. On the [CSM Compliance page][6], click **Create Framework**.
1. Enter the following details:
    - **Framework name**: The name of your framework. Can include characters, numbers, and spaces. Must be at least five characters long.

>     - **Handle**: The tag name for the custom framework. Can include lowercase letters, numbers, dashes, underscores, and periods. This value will be used for querying the framework in the explorer or dashboards.

Reviewer: ```suggestion
    - **Handle**: The tag name for the custom framework. Can include lowercase letters, numbers, dashes, underscores, and periods. This value is used to query the framework in the explorer or in dashboards.
```

    - **Version**: The version of the framework. Can include lowercase letters, numbers, dashes, underscores, and periods.

>     - **Image URL**: A publicly accessible URL for an image that will be used to identify the framework.

Reviewer: ```suggestion
    - **Image URL**: A publicly accessible URL for an image that is used to identify the framework.
```

1. Click **Next Step: Create Your Framework**.

Next, add requirements to the framework:

> <div class="alert alert-warning">You must add at least one requirement, control, and rule before you can save the custom framework.</div>

Reviewer: ```suggestion
<div class="alert alert-warning">You must add at least one requirement, control, and rule before saving the custom framework.</div>
```

1. Click **Add Requirement**.
1. Enter the following details:
    - **Requirement**: A requirement acts as a control family, enabling you to add controls and associate rules with each control.

>     - **Control**: A control represents the criteria that the requirement must meet and includes the rules associated with these criteria.

Author: Thanks, I'll add that to the PR.
